app/controllers/two-factor-auth-controller/post-configure-controller.js

```diff
@@ -2,14 +2,15 @@
 
 // NPM dependencies
 const logger = require('winston')
+const lodash = require('lodash')
 
 // Local dependencies
 const paths = require('../../paths')
 const userService = require('../../services/user_service.js')
 
 module.exports = (req, res) => {
   const code = req.body['code'] || ''
-  userService.configureNewOtpKey(req.user.externalId, code, 'APP', req.correlationId)
+  userService.configureNewOtpKey(req.user.externalId, code, lodash.get(req, 'session.pageData.twoFactorAuthMethod', 'APP'), req.correlationId)
```

Comment:
This is the critical change, we used to assume it was always `APP` now, it can be `APP || SMS`